Make June 2024 news public (#28009)
idaholab/moose: modules/doc/content/newsletter/2024/2024_06.md

# MOOSE Newsletter (June 2024)

## MOOSE Improvements

> 

Author: ```suggestion
```

### Using Physics on Components

> A new component, [FileMeshPhysicsComponent.md], was created in the thermal hydraulics module to leverage the [Physics](Physics/index.md) system. This facilitates the use of multi-dimensional equations in a thermal-hydraulics simulations, including notably the finite volume computational fluid dynamics capabilities of the Navier Stokes module.

Author: ```suggestion
A new component, [FileMeshPhysicsComponent.md], was created in the thermal hydraulics module to leverage the [Physics](Physics/index.md) system. This facilitates the use of multi-dimensional equations in a thermal hydraulics simulation including, notably, the finite volume computational fluid dynamics capabilities of the Navier Stokes module.
```

### Facilitating the use of mixture models

> The [WCNSFVTwoPhaseMixturePhysics.md] was added to the Navier Stokes module. This facilitates the set up of fully coupled finite volume simulations using this simple two phase mixture model. It can be used for both solidification and liquid mixture problems.

Author: ```suggestion
The [WCNSFVTwoPhaseMixturePhysics.md] was added to the Navier Stokes module. This facilitates the setup of fully coupled finite volume simulations using a simple two phase mixture model. It can be used for both solidification and liquid mixture problems.
```

> ## libMesh-level Changes
> 
> ## PETSc-level Changes
> 

Author: ```suggestion
```

## Bug Fixes and Minor Enhancements

> - Support for finite volume variables was added to the [ProjectionAux.md]
> - Support for specifying functors in parsed expressions was added to the [ParsedAux.md]
> - Subdomain names can now be pre-declared in the `[Mesh]` block, for subdomains that will be created during the simulation but do not exist at initialization. Objects can be block-restricted to these blocks at initialization.
> - Sideset ids and names can now be pre-declared in the `[Mesh]` block, for sidesets that will be created during the simulation but do not exist at initialization. Objects can be boundaries-restricted to these boundaries at initialization.
> - The [CombinerGenerator.md] now has the option to avoid merging subdomains and sidesets with the same ids in the meshes that it combines. When these options are selected, the ids are offset based on the maximum ids used in the meshes to combine with.

Author: ```suggestion
- Support for finite volume variables was added to [ProjectionAux.md]
- Support for specifying functors in parsed expressions was added to [ParsedAux.md]
- Subdomain names can now be pre-declared in the `[Mesh]` block, for subdomains that will be created during the simulation but do not exist at initialization. Objects can be block-restricted to these blocks at initialization.
- Sideset IDs and names can now be pre-declared in the `[Mesh]` block, for sidesets that will be created during the simulation but do not exist at initialization. Objects can be boundaries-restricted to these boundaries at initialization.
- The [CombinerGenerator.md] now has the option to avoid merging subdomains and sidesets with the same IDs in the meshes that it combines. When these options are selected, the IDs are offset based on the maximum ID used in the meshes being combined.
```
